### PR TITLE
Fix lem build in Makefile

### PR DESCRIPTION
**Pull request by Alasdair**
'class' is a reserved keyword in lem, so the use of class as a variable name was causing issues. While ultimately this should be fixed in Sail this can easily be worked around here.

Fortunately the code in questions was always using a pattern like
```
let class = f(x);
let y = match class {
  ...
};
y
```
which can be simplified to
```
match f(x) {
  ...
}
```
So I didn't have to invent a new variable name, and got to make the code simpler in the process!

Fixing this unfortunately causes some other issues to become apparent. The machine words library can't handle the large vlenmax, so we have to create a separate file with a smaller vlenmax for the theorem provers. There were also some other issues, so I switched out the default representation to bitlists until they can be resolved, and cleaned up some legacy cruft in handwritten_support while I was there.

The rmem target has similar issues, but as this is only really useful for rmem maintainers it doesn't make sense to run by default, so I removed it from the default set of targets.

This should fix https://github.com/riscv/sail-riscv/issues/325

### CODE REVIEW

**Comment by Alasdair** [comment]
I split the commits so the actual model changes are in the first commit (which are all small refactorings).

The second commit contains all the lem hackery to fix the build

File: model/riscv_insts_vext_utils.sail [hunk 1]
```diff
@@ -976,8 +976,7 @@ function rsqrt7 (v, sub) = {
 
 val riscv_f16Rsqrte7 : (bits_rm, bits_H) -> (bits_fflags, bits_H) effect {rreg}
```
Comment (billmcspadden-riscv) [inline model/riscv_insts_vext_utils.sail:977]: Remove effect throughout.

File: model/riscv_insts_vext_utils.sail [hunk 2]
```diff
@@ -1006,14 +1003,12 @@ function riscv_f32Rsqrte7 (rm, v) = {
     0x0080 => (zeros(5), 0x00000000),
     0x0020 => (zeros(5), rsqrt7(v, true)[31 .. 0]),
     _      => (zeros(5), rsqrt7(v, false)[31 .. 0])
-  };
-  (fflags, result)
+  }
 }
 
 val riscv_f64Rsqrte7 : (bits_rm, bits_D) -> (bits_fflags, bits_D) effect {rreg}
```
Comment (billmcspadden-riscv) [inline model/riscv_insts_vext_utils.sail:1009]: Remove effect

File: model/riscv_insts_vext_utils.sail [hunk 3]
```diff
@@ -988,14 +987,12 @@ function riscv_f16Rsqrte7 (rm, v) = {
     0x0080 => (zeros(5), 0x0000),
     0x0020 => (zeros(5), rsqrt7(v, true)[15 .. 0]),
     _      => (zeros(5), rsqrt7(v, false)[15 .. 0])
-  };
-  (fflags, result)
+  }
 }
 
 val riscv_f32Rsqrte7 : (bits_rm, bits_S) -> (bits_fflags, bits_S) effect {rreg}
```
Comment (billmcspadden-riscv) [inline model/riscv_insts_vext_utils.sail:993]: Remove effect.

File: model/riscv_vlen256.sail
Comment (billmcspadden-riscv) [inline model/riscv_vlen256.sail:1]: Why is this part of this PR?  Seems like an orthogonal addition.

**Review by billmcspadden-riscv — COMMENTED** [review]
My review is mainly questions.  I don't see anything wrong in the code and will approve the PR once the questions are answered.

One exception:  the 'effects' of certain functions need to be removed, yes?

Unable to comment on lem code due to lack of knowledge.

File: Makefile [hunk 4]
```diff
@@ -184,15 +185,7 @@ C_LIBS += $(SAIL_LIB_DIR)/coverage/libsail_coverage.a -lm -lpthread -ldl
 endif
 
 RISCV_EXTRAS_LEM_FILES = riscv_extras.lem mem_metadata.lem riscv_extras_fdext.lem
-# Feature detect if we are on the latest development version of Sail
```
[the remark this answers is not in the document]
Comment (Alasdair) [reply Makefile:187]: It's a check for a version of Sail that is many years old, and no longer works with the model.

File: model/riscv_insts_vext_utils.sail (hunk 1 above)
Comment (Alasdair) [reply model/riscv_insts_vext_utils.sail]: These lines are not part of this PR

Comment (Alasdair) [reply model/riscv_insts_vext_utils.sail]: You are seeing them just because this PR was opened before my other PR removing the effects separately was merged.

**Comment by jrtc27** [comment]
> Merge branch 'master' into class_fix

Please rebase rather than merge to keep the history untangled

File: model/riscv_vlen256.sail
Comment (Alasdair) [reply model/riscv_vlen256.sail:1]: Ideally we would be able to disable V, but the extension is too intertwined into the base model right now to have a sail-riscv model without V, so having a shorter vlenmax is the best we can do.

**Comment by Alasdair** [comment]
> > Merge branch 'master' into class_fix
> 
> Please rebase rather than merge to keep the history untangled

Sorry, I didn't realise the github update branch button did a merge by default. I'll fix it.

**Comment by bacam** [comment]
For what it's worth, I've got development versions of Lem and Sail to handle the model with vectors and machine words, but I think the switch to bitlists by default is worthwhile because it's a little more robust.  However, this does mean that the `vlenmax` change is unnecessary, so I suggest dropping that.

**Comment by Alasdair** [comment]
Ok, the additional vlen file is gone and it seems to build fine.

**Comment by Alasdair** [comment]
I think more people are running into this issue, would be good to get this merged I think

**Comment by billmcspadden-riscv** [comment]
Will do
